lib/simple-capistrano-unicorn/namespace.rb

```diff
@@ -82,63 +67,122 @@ def clean_old_unicorn(server)
+          # It's following the: http://unicorn.bogomips.org/SIGNALS.html
+          #
           desc "Zero-downtime restart of Unicorn"
           task :restart do
+            STDOUT.sync
+
+            # 0. PRINT STATUS OF ALL APP-SERVERS AND UNICORN STATUTES
+            print "Deploying to these app-servers:\n".colorize(:light_white).bold
             find_servers(:roles => :app).each do |server|
               if unicorn_is_running?(server)
-                pid = capture "cat #{unicorn_pid}", :hosts => [server]
-                run "kill -s USR2 #{pid.to_i}", :hosts => [server] if pid.to_i > 0
-                logger.info nice_output("Restarted Unicorn!", server)
+                print "♞".colorize(:green)
               else
-                logger.info nice_output("Unicorn wasn't running, starting it!", server)
+                print "♞".colorize(:red)
+              end
+              print " -#{server.host}\n".colorize(:white)
+            end
+
+            # 1. MAKES SURE ALL SERVERS ARE RUNNING!
+            print "Making sure all servers are running".colorize(:light_white).bold
+            print ".......|".colorize(:white)
+            find_servers(:roles => :app).each do |server|
+              unless unicorn_is_running?(server)
                 start_unicorn(server)
                 logger.info nice_output("Started Unicorn!", server)
+                print ".".colorize(:yellow)
+              else
+                print ".".colorize(:green)
               end
             end
-            # Only clean-up if unicorn don't kill its old master
-            unicorn.cleanup unless unicorn_suicide
+            print "✓\n".colorize(:green).bold
+
+            # 2. MAKE ALL SERVERS RELOAD THE NEW CODE!
+            print "Reloading new code (USR2)".colorize(:light_white).bold
+            print ".................|".colorize(:white)
+            sleep(10)
+            find_servers(:roles => :app).each do |server|
+              pid = capture("cat #{unicorn_pid}", :hosts => [server]).to_i
+              run "kill -s USR2 #{pid}", :hosts => [server]
+              print ".".colorize(:green)
+            end
+            print "✓\n".colorize(:green).bold
+
+            # 3. MAKE ALL SERVERS STOP SENDING TRAFFIC TO OLD MASTER
+            print "Killing workers from old masters (WINCH)".colorize(:light_white).bold
+            print "..|".colorize(:white)
+            sleep(10)
+            find_servers(:roles => :app).each do |server|
+              old_pid = capture("cat #{unicorn_old_pid}", :hosts => [server]).to_i
+              run "kill -s WINCH #{old_pid}", :hosts => [server]
+              print ".".colorize(:green)
+            end
+            print "✓\n".colorize(:green).bold
+
+            # 3.1 ALL TRAFFIC GOES TO NEW SERVERS, WANT TO KILL OLD?
+            while(true)
+              print "Traffic is handled by new master, are you happy (y/n)? ".colorize(:white)
+              answer = STDIN.gets.downcase.gsub("\n",'')
```

Comment:
Probably a good idea to have support for auto-answer, to specify in setting if it always should default to `Y` or ask each time. By the current way, it loosing it's nice, clean and quick deploys.